<launch>
  <arg name="INPUT_IMAGE" default="/multisense/left/image_rect_color"/>
  <arg name="INPUT_CAMERA_INFO" default="/multisense/left/camera_info"/>
  <arg name="INPUT_POINT_CLOUD" default="/multisense/resize_1_4/points" />
  <arg name="DEBUG" default="false" />
  <group ns="valve_recognition">
    <node pkg="image_view2" type="image_view2" name="image_view2" if="$(arg DEBUG)">
      <remap from="image" to="$(arg INPUT_IMAGE)" />
      <remap from="$(arg INPUT_IMAGE)/screenrectangle" to="input_rect" />
    </node>
    <node pkg="jsk_perception" type="rect_to_mask_image"
        name="rect_to_mask_image" output="screen">
    <remap from="~input" to="input_rect" />
    <remap from="~input/camera_info" to="$(arg INPUT_CAMERA_INFO)" />
    </node>
    <node pkg="jsk_pcl_ros" type="mask_image_filter" name="mask_image_filter">
      <remap from="~input" to="$(arg INPUT_POINT_CLOUD)"/>
      <remap from="~input/mask" to="rect_to_mask_image/output"/>
      <remap from="~input/camera_info" to="$(arg INPUT_CAMERA_INFO)" />
    </node>
    <node pkg="nodelet" type="nodelet" name="filtered_cloud"
          args="standalone pcl/ExtractIndices">
      <remap from="~input" to="$(arg INPUT_POINT_CLOUD)"/>
      <remap from="~indices" to="mask_image_filter/output"/>
    </node>

    <node pkg="jsk_pcl_ros" type="torus_finder" name="valve_finder">
      <remap from="~input" to="filtered_cloud/output" />
      <rosparam>
        max_iterations: 1000
        min_radius: 0.05
      </rosparam>
    </node>
    <node pkg="jsk_topic_tools" type="snapshot" name="snapshot">
      <remap from="~input" to="valve_finder/output"/>
      <rosparam>
        latch: true
      </rosparam>
    </node>
  </group>
</launch>
